src/Features/Core/Portable/CodeFixes/NamingStyle/AbstractNamingStyleCodeFixProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.NamingStyles
+{
+    internal abstract class AbstractNamingStyleCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(IDEDiagnosticIds.NamingRuleId); }
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var serializedNamingStyle = diagnostic.Properties[nameof(NamingStyle)];
+
+            NamingStyle style;
+            using (var reader = XmlReader.Create(new StringReader(serializedNamingStyle)))
+            {
+                var serializer = new DataContractSerializer(typeof(NamingStyle));
+                style = serializer.ReadObject(reader) as NamingStyle;
+            }
+
+            var document = context.Document;
+            var span = context.Span;
+
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindToken(span.Start).GetAncestors<SyntaxNode>().First(n => n.Span.Contains(span));
+            var model = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            var fixedNames = style.MakeCompliant(symbol.Name);
```

Comment:
When is there more than name to fix.  I ask because it would be great to move some of this logic into the lambda inside `RegisterCodeFix`.  Everything directly in this method runs on "every" caret move to determine whether or not to show a lightbulb, and this seems to be quite lot.  Would be nice to defer as much as possible.